addons/binding/org.openhab.binding.ihc/src/main/java/org/openhab/binding/ihc/internal/ws/IhcClient.java

```diff
@@ -0,0 +1,558 @@
+        private boolean interrupted = false;
+
+        public void setInterrupted(boolean interrupted) {
+            this.interrupted = interrupted;
+            this.interrupt();
+        }
+
+        @Override
+        public void run() {
+            logger.debug("IHC controller state listener started");
+            WSControllerState previousState = null;
+
+            // as long as no interrupt is requested, continue running
+            while (!interrupted) {
+                try {
+                    if (previousState == null) {
+                        // initialize previous state
+                        previousState = controllerService.getControllerState();
+                        logger.debug("Controller initial state {}", previousState.getState());
+                    }
+                    logger.trace("Wait new state change notification from controller");
+                    WSControllerState currentState = waitStateChangeNotifications(previousState,
+                            NOTIFICATION_WAIT_TIMEOUT_IN_SEC);
+                    logger.trace("Controller state {}", currentState.getState());
+
+                    if (previousState.getState().equals(currentState.getState()) == false) {
+                        logger.debug("Controller state change detected ({} -> {})", previousState.getState(),
+                                currentState.getState());
+                        sendControllerStateUpdateEvent(currentState);
+                        previousState.setState(currentState.getState());
+                    }
+                } catch (IhcExecption e) {
+                    if (!interrupted) {
+                        logger.error("New controller state change notification wait failed...", e);
+                        sendErrorEvent(e);
+                        mysleep(1000L);
+                    }
+                }
+            }
+            logger.debug("IHC controller state listener stopped");
+        }
+
+        private void mysleep(long milli) {
+            try {
+                sleep(milli);
+            } catch (InterruptedException e) {
+                interrupted = true;
+            }
+        }
+    }
+
+    private void sendErrorEvent(IhcExecption err) {
+        try {
+            Iterator<IhcEventListener> iterator = eventListeners.iterator();
+
+            while (iterator.hasNext()) {
+                iterator.next().errorOccured(err);
+            }
+        } catch (Exception e) {
+            logger.error("Event listener invoking error", e);
+        }
+    }
+
+    private void sendControllerStateUpdateEvent(WSControllerState state) {
+        try {
+            Iterator<IhcEventListener> iterator = eventListeners.iterator();
+
+            while (iterator.hasNext()) {
+                iterator.next().statusUpdateReceived(state);
+            }
+        } catch (Exception e) {
+            logger.error("Event listener invoking error", e);
```

Comment:
Don't log to error it's for system wide failure only.

Reply:
Done